### ISSUE TITLE
TypeError: "paths[0]" property must be of type string, got undefined

### ISSUE THREAD

**Issue by jackra1n** [huntabyte/shadcn-svelte#1003]
### Describe the bug

When trying to init shadcn-svelte with bun im getting the following error:
```
┌   shadcn-svelte  v0.10.1 
│
└  TypeError: "paths[0]" property must be of type string, got undefined
    at join (native)
    at Zs (/tmp/bunx-1000-shadcn-svelte@latest/node_modules/shadcn-svelte/dist/index.js:1088:20)
    at Ys (/tmp/bunx-1000-shadcn-svelte@latest/node_modules/shadcn-svelte/dist/index.js:1071:19)
    at <anonymous> (/tmp/bunx-1000-shadcn-svelte@latest/node_modules/shadcn-svelte/dist/index.js:1098:45)
    at processTicksAndRejections (:12:39)
```
### Reproduction

```
bun create svelte@latest myApp

-> (select Skeleton project, TypeScript, ESLint, Prettier)

cd myApp
bun install
bun pm trust --all
bunx svelte-add@latest tailwindcss
bun install
bunx shadcn-svelte@latest init
```

### System Info

```bash
System:
    OS: Linux 6.8 Fedora Linux 39 (Workstation Edition)
    CPU: (24) x64 AMD Ryzen 9 5900X 12-Core Processor
    Memory: 6.50 GB / 31.26 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    bun: 1.1.3 - /tmp/bun-node-2615dc742/bun
  Browsers:
    Chrome: 123.0.6312.105
  npmPackages:
    @sveltejs/kit: ^2.0.0 => 2.5.5 
    svelte: ^4.2.7 => 4.2.12
```

### Severity

blocking all usage of shadcn-svelte

**Comment by Triips-TheCoder** [comment]
Got the same bug with `pnpm` it's blocking totally the usage of `shadcn-svelte`. 

<img width="1636" alt="Capture d’écran 2024-04-10 à 10 02 56" src="https://github.com/huntabyte/shadcn-svelte/assets/73161461/ea17bc22-1934-405e-a565-b2462de179d2">

**Comment by jhakeinson** [comment]
I am also getting the same error. I'm using bun.

**Comment by AdrianGonz97** [comment]
@jackra1n Thank you for the reproduction! Unfortunately, I'm unable to reproduce the issue with the provided steps. Could you provide a demo repo so that I can have a concrete project to work off of?

---

@Triips-TheCoder, @jhakeinson Thanks for the reports. 
Can you both run: 
```bash
npx envinfo --system --npmPackages svelte,shadcn-svelte,@sveltejs/kit --binaries --browsers
``` 
in the projects that you're having this issue with and provide the logged output here? This is mainly to see if there's a common thread among systems.

**Comment by Triips-TheCoder** [comment]
> @jackra1n Thank you for the reproduction! Unfortunately, I'm unable to reproduce the issue with the provided steps. Could you provide a demo repo so that I can have a concrete project to work off of?
> 
> **Edit:** It looks like I _can_ reproduce it if I use the `--bun` flag in `bunx --bun shadcn-svelte@0.10.1 init`. If this is the case for you, then I would suggest to not do that. This forces the script to run in the Bun runtime, which completely disregards the shebang that we specify at the start of the CLI script. We don't actively test or support the compatibility of other runtimes besides Node.
> 
> @jhakeinson Thanks for the report. Can you run:
> 
> ```shell
> npx envinfo --system --npmPackages svelte,shadcn-svelte,@sveltejs/kit --binaries --browsers
> ```
> 
> in the projects that you're having this issue with and provide the logged output here? This is mainly to see if there's a common thread among systems.
> 
> @Triips-TheCoder Looking at the error, this seems to be unrelated to this issue. Can you raise a separate issue with a reproduction?

I checked and you're right ! It was an issue with my `node_modules` folder. After deleted it then retry the command everything worked fine thank you.

**Comment by jackra1n** [comment]
> @jackra1n Thank you for the reproduction! Unfortunately, I'm unable to reproduce the issue with the provided steps. Could you provide a demo repo so that I can have a concrete project to work off of?
> 
> **Edit:** It looks like I _can_ reproduce it if I use the `--bun` flag in `bunx --bun shadcn-svelte@0.10.1 init`. If this is the case for you, then I would suggest to not do that. This forces the script to run in the Bun runtime, which completely disregards the shebang that we specify at the start of the CLI script. We don't actively test or support the compatibility of other runtimes besides Node.

I am not adding `--bun` to my command. I do not have Node installed on my system

**Comment by AdrianGonz97** [comment]
> I am not adding --bun to my command. I do not have Node installed on my system

The result is the same then. The CLI is being executed in the Bun runtime (as Node isn't present), which is not supported. We support the Bun package manager, but not the runtime.

To fix, install Node and bunx will execute the CLI script in the correct runtime.

**Comment by AdrianGonz97** [comment]
#1011 Fixes this issue and unblocks Bun and Deno from being used to run the CLI. 

With that said, runtimes outside of Node v18 or higher **are not officially supported**. The fix in this case was trivial, but future issues may not be. As such, if you want a stable and consistent experience, just stick to Node. Using Bun as a package manager should be fine, though.
